### ISSUE TITLE
CdkTrapFocus uses deprecated FocusTrapFactory

### ISSUE THREAD

**Issue by pfeileon** [angular/components#21703]
Extending `CdkTrapFocus` causes the issue that its dependency `FocusTrapFactory` is deprecated (and should have actually been removed already: `@breaking-change: for 11.0.0 Remove this class`) and its proposed substitution `ConfigurableFocusTrapFactory` is superficially incompatible with `CdkTrapFocus` (it seems to work by simply suppressing the compiler with `// @ts-ignore`).

What is the preferred way to deal with this issue?

**Comment by eehaddad** [comment]
Also interested in this, if anyone has any insight on how to use the directive with a deprecated dependency or any alternatives.

**Comment by playnox** [comment]
+1

**Comment by KutskoAnton** [comment]
+1

**Comment by crisbeto** [comment]
Closing since `FocusTrapFactory` isn't marked as deprecated anymore.
